pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -592,14 +642,116 @@ func (l *listUsersQuery) expandExclusion(
 	}
 
 	wildcardKey := tuple.TypedPublicWildcard(req.GetUserFilters()[0].GetType())
-	_, subtractWildcardExists := subtractFoundUsersMap[wildcardKey]
-	for key := range baseFoundUsersMap {
-		if _, isSubtracted := subtractFoundUsersMap[key]; !isSubtracted && !subtractWildcardExists {
-			// Iterate over base users because at minimum they need to pass
-			// but then they are further compared to the subtracted users map.
-			// If users exist in both maps, they are excluded. Only users that exist
-			// solely in the base map will be returned.
-			trySendResult(ctx, tuple.StringToUserProto(key), foundUsersChan)
+
+	_, baseWildcardExists := baseFoundUsersMap[wildcardKey]
+	subtractWildcardUser, subtractWildcardExists := subtractFoundUsersMap[wildcardKey]
+
+	if baseWildcardExists {
+		if !subtractWildcardExists || subtractWildcardUser.noRelationship {
+			trySendResult(ctx, foundUser{
+				user: tuple.StringToUserProto(wildcardKey),
+			}, foundUsersChan)
+		}
+
+		residualSubtractedUserMap := maps.Clone(subtractFoundUsersMap)
```

Comment:
Is this clone necessary? Ideally the `subtractFoundUsersMap` shouldn't be mutated.

Reply:
It's no longer necessary with the new control flow pattern. I've removed it 👍